tool/tsh/common/git.go

```diff
@@ -19,24 +19,119 @@
 package common
 
 import (
+	"bytes"
+	"io"
+	"os/exec"
+	"strings"
+
 	"github.com/alecthomas/kingpin/v2"
+	"github.com/go-git/go-git/v5/plumbing/transport"
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/types"
 )
 
 type gitCommands struct {
-	list  *gitListCommand
-	login *gitLoginCommand
+	list   *gitListCommand
+	login  *gitLoginCommand
+	ssh    *gitSSHCommand
+	config *gitConfigCommand
+	clone  *gitCloneCommand
 }
 
 func newGitCommands(app *kingpin.Application) gitCommands {
 	git := app.Command("git", "Git server commands.")
 	cmds := gitCommands{
-		login: newGitLoginCommand(git),
-		list:  newGitListCommand(git),
+		login:  newGitLoginCommand(git),
+		list:   newGitListCommand(git),
+		ssh:    newGitSSHCommand(git),
+		config: newGitConfigCommand(git),
+		clone:  newGitCloneCommand(git),
 	}
 
 	// TODO(greedy52) hide the commands until all basic features are implemented.
 	git.Hidden()
 	cmds.login.Hidden()
 	cmds.list.Hidden()
+	cmds.config.Hidden()
+	cmds.clone.Hidden()
 	return cmds
 }
+
+type gitSSHURL struct {
+	*transport.Endpoint
```

Comment:
nit: my personal preference is to name each field explicitly to avoid confusion, e.g.:

```suggestion
	Endpoint *transport.Endpoint
```

Again, just my personal view, feel free to ignore.

Reply:
ended up like this:
```go
type gitSSHURL transport.Endpoint
```
there is no need for other attributes (yet).